### PR TITLE
Add error view for course search

### LINKED ISSUE #19: Better error view for courses that does not exist or not offered this term

### PR DESCRIPTION
**Pull request by 19506jk**
Include network error and course not found error
Improve error handling when fetching for a course
Prevent user from cancelling the loading dialog
Resolve #19

### CODE REVIEW

File: app/src/main/java/com/YC2010/jason/myclass/data/Connections.java [hunk 1]
```diff
@@ -155,9 +155,8 @@ public static JSONObject getJSON_from_url(String url) throws Exception {
 
             int responseCode = connection.getResponseCode();
 
-            if (null == connection.getInputStream()) return null;
```
Comment (IYCI) [inline app/src/main/java/com/YC2010/jason/myclass/data/Connections.java:158]: no need to handle null == connection.getInputStream situation?

File: app/src/main/java/com/YC2010/jason/myclass/data/Connections.java [hunk 2]
```diff
@@ -155,9 +155,8 @@ public static JSONObject getJSON_from_url(String url) throws Exception {
 
             int responseCode = connection.getResponseCode();
 
-            if (null == connection.getInputStream()) return null;
-
             if (responseCode != 200) {
+                Log.d("getJSONObject", "Failed: HTTP error code: " + responseCode);
                 throw new RuntimeException("Failed: HTTP error code: " + responseCode);
```
Comment (IYCI) [inline app/src/main/java/com/YC2010/jason/myclass/data/Connections.java:159]: why add a log?

File: app/src/main/java/com/YC2010/jason/myclass/data/fetchtasks/SearchFetchTask.java [hunk 3]
```diff
@@ -29,26 +29,27 @@
...
-        mAsyncTaskCallbackInterface = asyncTaskCallbackInterface;
-    }
-
     private ArrayList<LectureSectionObject> lectures = new ArrayList<>();
     private ArrayList<TutorialObject> tutorials = new ArrayList<>();
     private ArrayList<TestObject> tests = new ArrayList<>();
     private ArrayList<FinalObject> finals = new ArrayList<>();
 
-    ProgressDialog progDailog;
+    private ProgressDialog progDailog;
```
Comment (IYCI) [inline app/src/main/java/com/YC2010/jason/myclass/data/fetchtasks/SearchFetchTask.java]: typo, dialog not dailog

File: app/src/main/java/com/YC2010/jason/myclass/ui/fragments/SearchFragment.java [hunk 4]
```diff
@@ -38,7 +45,39 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+                errorIcon.setImageResource(R.drawable.ic_cloud_off_black_36dp);
+                return mView;
+            }
+
+            SearchFetchTask searchFetchTask = new SearchFetchTask(mActivity, new AsyncTaskCallbackInterface() {
+                @Override
+                public void onOperationComplete(Bundle bundle) {
+                    if (bundle != null && bundle.getBoolean("valid_return")) {
+                        setupTabs(bundle);
+                        LinearLayout tabLayout = (LinearLayout) mView.findViewById(R.id.search_detail_layout);
```
Comment (IYCI) [inline app/src/main/java/com/YC2010/jason/myclass/ui/fragments/SearchFragment.java]: the name tabLayout is confusing, why not use searchLayout

File: app/src/main/java/com/YC2010/jason/myclass/ui/fragments/SearchFragment.java [hunk 5]
```diff
@@ -38,7 +45,39 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+                        tabLayout.setVisibility(View.VISIBLE);
+                    } else {
+                        LinearLayout errorView = (LinearLayout) mView.findViewById(R.id.error_view);
```
Comment (IYCI) [inline app/src/main/java/com/YC2010/jason/myclass/ui/fragments/SearchFragment.java]: can we write a method for displaying error view

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -77,5 +77,8 @@
     <string name="course_tab_title1">Schedule</string>
     <string name="course_tab_title2">Exam</string>
     <string name="course_tab_title3">Info</string>
+    <string name="error_message_network">There is no network connection right now</string>
+    <string name="error_message_course">Course is not available this term or it does not exist</string>
+    <string name="error_retry_button">try again</string>
```
Comment (IYCI) [inline app/src/main/res/values/strings.xml:82]: how did u make it appears all upper case, just wondering

File: app/src/main/res/layout/fragment_search.xml [hunk 7]
```diff
@@ -1,26 +1,71 @@
...
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_centerInParent="true"
+                android:layout_toEndOf="@id/error_icon"
+                android:text="ttttt"
+                android:textSize="16sp" />
+        </RelativeLayout>
+
+        <Button
+            android:id="@+id/button_send"
```
Comment (IYCI) [inline app/src/main/res/layout/fragment_search.xml]: try name like error_view_layout, error_view_icon, error_view_message and most importantly error_view_button.

File: app/src/main/java/com/YC2010/jason/myclass/ui/fragments/SearchFragment.java [hunk 8]
```diff
@@ -38,7 +45,39 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+                        TextView errorMessage = (TextView) mView.findViewById(R.id.error_message);
```
Comment (IYCI) [inline app/src/main/java/com/YC2010/jason/myclass/ui/fragments/SearchFragment.java]: in this condition, need to set the retry button visibility gone or something like that.

File: app/src/main/java/com/YC2010/jason/myclass/ui/fragments/SearchFragment.java [hunk 9]
```diff
@@ -38,7 +45,39 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         if (mView == null) {
             mView = inflater.inflate(R.layout.fragment_search, container, false);
+
+            if (!Connections.isNetworkAvailable(mActivity)) {
+                LinearLayout errorView = (LinearLayout) mView.findViewById(R.id.error_view);
+                TextView errorMessage = (TextView) mView.findViewById(R.id.error_message);
+                ImageView errorIcon = (ImageView) mView.findViewById(R.id.error_icon);
+
```
Comment (IYCI) [inline app/src/main/java/com/YC2010/jason/myclass/ui/fragments/SearchFragment.java]: if we are displaying the retry button, better implement it. or we can hide it and implement it later.(open an issue for it, nice to have feature)

File: app/src/main/java/com/YC2010/jason/myclass/data/Connections.java (hunk 1 above)
Comment (19506jk) [inline app/src/main/java/com/YC2010/jason/myclass/data/Connections.java:158]: if the status is 200 then the school server should return something

**Pull request merged**